yascheduler/clouds/hetzner.py

```diff
@@ -40,22 +38,17 @@ def create_node(self):
         response = self.client.servers.create(
             name=self.get_rnd_name('node'),
             server_type=ServerType('cx51'), image=Image(name='debian-10'),
-            ssh_keys=[SSHKey(name=self.key_name)]
+            ssh_keys=[SSHKey(name=self.key_name)],
+            user_data="#cloud-config\n" + json.dumps(self.cloud_config_data),
         )
         server = response.server
         ip = server.public_net.ipv4.ip
         logging.info('CREATED %s' % ip)
 
-        time.sleep(5)
-
-        # warm up
-        for _ in range(10):
-            ssh_conn = SSH_Connection(
-                host=ip, user=self.ssh_user, connect_kwargs=self.ssh_custom_key
-            )
-            try: ssh_conn.run('whoami', hide=True)
-            except: time.sleep(5)
-            else: break
+        # wait node up and ready
+        self._run_ssh_cmd_with_backoff(
+            ip, cmd="cloud-init status --wait", max_wait_time=600
```

Comment:
@knopki should we indeed wait for 600 seconds instead of about 55 seconds? :thinking:

Reply:
@blokhin I think it's ok to wait up to 600 seconds with up to 60 retries every 10 seconds. But the problem is that I misplaced arguments.
Fixed. Up to 60 seconds (retries every 5 seconds) for Hetzner and up to 600 seconds (retries every 10 seconds) for slow-as-hell Azure. Practically, SSH connection is successful on first or second try.